[[ Bug 23176 ]] Ensure the target MacOS SDK (10.9) is properly set

livecodepanos opened the pull request:
This patch ensures that hardened runtime will not block externals from launching. This happened because the target sdk version was not set in the externals after building with Xcode12+.

Note: Xcode uses clang to drive ld - `-Wl`, means pass this option to the linker direct

livecodepanos commented:
Submitting this on behalf of @runrevmark

runrevmark commented on config/mac.gypi:
```diff
@@ -30,7 +30,14 @@
 		'CLANG_LINK_OBJC_RUNTIME': 'NO',
 		'CLANG_CXX_LANGUAGE_STANDARD': 'c++11',
-        'CLANG_CXX_LIBRARY': 'libc++'
+        'CLANG_CXX_LIBRARY': 'libc++',
```
Use tabs for indent here - just to make it line up with the options above.

livecodepanos commented:
WIP since this fails on Travis, with the following error:
```
❌  ld: unknown option: -platform_version
157
158
159
160❌  clang: error: linker command failed with exit code 1 (use -v to see invocation)
161
162
163** BUILD FAILED **
164
165
166The following build commands failed:
167	Ld /Users/travis/build/livecode/livecode/_build/mac/Debug/perfect-target normal x86_64
168(1 failure)
169make[1]: *** [compile-mac] Error 65
170make: *** [all-mac] Error 2
```

livecodepanos committed:
[[ Bug 23176 ]] Ensure the target MacOS SDK (10.9) is properly set

This patch ensures that hardened runtime will not block externals from
launching. This happened because the target sdk version was not
set in the externals after building with Xcode12+.

Note: Xcode uses clang to drive ld - '-Wl', means pass this option to the linker direct

livecodepanos commented:
Tested locally

@livecode-vulcan review ok 4f0e8e2

livecode-vulcan commented:
💙 review by @livecodepanos ok 4f0e8e28b254beeb7f3bf686921817078089daa5

livecode-vulcan commented:
😎 test success 4f0e8e28b254beeb7f3bf686921817078089daa5

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/992)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/400)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/399)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/398)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/399)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2517)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/46)
* try-community-universal-ios-iphoneos14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.1/builds/95)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/46)
* try-community-universal-ios-iphonesimulator14.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.1/builds/93)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2544)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2533)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2531)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2568)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2504)
